src/lib/cache/index.js

```diff
@@ -0,0 +1,64 @@
+import fs from "fs";
+import path from "path";
+
+const PROMISES_CACHE_FILENAME = "promises.json";
+const SITES_CACHE_FILENAME = "sites.json";
+
+function getFilePath(filename) {
+  const publicDirectory = path.join(process.cwd(), "public/data");
+  return path.join(publicDirectory, filename);
+}
+
+function cache(server, fetchFor) {
+  const CACHE_TIMEOUT = Number.parseInt(server.env("CACHE_TIMEOUT"), 10);
+
+  async function read(filename) {
```

Comment:
Since we are reading the file synchronously, do we need to make this function async?

Reply:
I thought I made it async already ... Damn. Will look into it